### ISSUE TITLE
header addition

### ISSUE THREAD

**Issue by jmfriedt** [gpuocelot/gpuocelot#9]
I am sure this is obvious to anyone fluent in C++ but on the current Debian sid distribution I had to add ``#include <cstdint>`` in the headers included from ``ocelot/ThirdParty/hydrazine/include/hydrazine/json.h`` for ``gpuocelot`` to complete compilation.

**Comment by dmikushin** [comment]
Hi @jmfriedt , thank you for this report. Could you please confirm the below is needed:

```diff
diff --git a/include/hydrazine/json.h b/include/hydrazine/json.h
index 36719f5..d665356 100644
--- a/include/hydrazine/json.h
+++ b/include/hydrazine/json.h
@@ -11,6 +11,7 @@
 #ifndef HYDRAZINE_JSON_H_INCLUDED
 #define HYDRAZINE_JSON_H_INCLUDED
 
+#include <cstdint>
 #include <vector>
 #include <map>
 #include <string>
```

?

**Comment by jmfriedt** [comment]
yes correct. Without this header the ``uint64_t()`` was unknown.

Also I am not sure if relevant, but
```
.../build$ ctest
Test project /tmp/gpuocelot/ocelot/build
    Start 1: hydrazine_TestArgumentParser
1/6 Test #1: hydrazine_TestArgumentParser ......   Passed    0.00 sec
    Start 2: hydrazine_TestMath
2/6 Test #2: hydrazine_TestMath ................   Passed    0.07 sec
    Start 3: hydrazine_TestThread
3/6 Test #3: hydrazine_TestThread ..............   Passed    0.02 sec
    Start 4: hydrazine_TestTimer
4/6 Test #4: hydrazine_TestTimer ...............   Passed    0.00 sec
    Start 5: hydrazine_TestXmlArgumentParser
5/6 Test #5: hydrazine_TestXmlArgumentParser ...   Passed    0.01 sec
    Start 6: hydrazine_TestXmlParser
6/6 Test #6: hydrazine_TestXmlParser ...........   Passed    0.01 sec

100% tests passed, 0 tests failed out of 6

```
only runs much fewer tests than stated in the main README file.
Thanks.

**Comment by dmikushin** [comment]
Fixed in 342a9d782d983479dc74a64954b3c1527c939cf4 thanks!

**Comment by dmikushin** [comment]
> only runs much fewer tests than stated in the main README file.

Please make sure `-DOCELOT_BUILD_TESTS=ON` is given to CMake, because you missed all ocelot tests (and executed only hydrazine's).

**Comment by jmfriedt** [comment]
I did but
```
  Manually-specified variables were not used by the project:
    OCELOT_BUILD_TESTS

```

**Comment by dmikushin** [comment]
Sorry, it must be `BUILD_TESTS` instead of `OCELOT_BUILD_TESTS`

**Comment by jmfriedt** [comment]
right, I should have read the output of ``cmake-gui ../`` more carefully rather than following the README commands: running ``cmake ../ -DBUILD_TESTS=ON -DBUILD_TESTS_CUDA=OFF`` does generate all the tests with CUDA 12 shipped with the current Debian sid:
```
Test project /t/tmp/gpuocelot/ocelot/build
      Start  1: gpuocelot_analysis_TestDataflowGraph
 1/18 Test  #1: gpuocelot_analysis_TestDataflowGraph ...........   Passed    0.01 sec
      Start  2: gpuocelot_api_TestDeviceSwitching
 2/18 Test  #2: gpuocelot_api_TestDeviceSwitching ..............   Passed    0.02 sec
      Start  3: gpuocelot_api_TestExternalFunctions
 3/18 Test  #3: gpuocelot_api_TestExternalFunctions ............   Passed    0.03 sec
      Start  4: gpuocelot_executive_TestEmulator
 4/18 Test  #4: gpuocelot_executive_TestEmulator ...............   Passed    0.01 sec
      Start  5: gpuocelot_executive_TestInstructions
 5/18 Test  #5: gpuocelot_executive_TestInstructions ...........   Passed    0.01 sec
      Start  6: gpuocelot_executive_TestKernels
 6/18 Test  #6: gpuocelot_executive_TestKernels ................   Passed    0.02 sec
      Start  7: gpuocelot_executive_TestLLVMKernels
 7/18 Test  #7: gpuocelot_executive_TestLLVMKernels ............   Passed    0.08 sec
      Start  8: gpuocelot_ir_TestLLVMInstructions
 8/18 Test  #8: gpuocelot_ir_TestLLVMInstructions ..............   Passed    0.01 sec
      Start  9: gpuocelot_parser_TestLexer
 9/18 Test  #9: gpuocelot_parser_TestLexer .....................   Passed    0.01 sec
      Start 10: gpuocelot_parser_TestParser
10/18 Test #10: gpuocelot_parser_TestParser ....................   Passed    0.01 sec
      Start 11: gpuocelot_transforms_TestOptimizations
11/18 Test #11: gpuocelot_transforms_TestOptimizations .........   Passed    0.01 sec
      Start 12: gpuocelot_translator_TestPTXToLLVMTranslator
12/18 Test #12: gpuocelot_translator_TestPTXToLLVMTranslator ...   Passed    0.01 sec
      Start 13: hydrazine_TestArgumentParser
13/18 Test #13: hydrazine_TestArgumentParser ...................   Passed    0.00 sec
      Start 14: hydrazine_TestMath
14/18 Test #14: hydrazine_TestMath .............................   Passed    0.07 sec
      Start 15: hydrazine_TestThread
15/18 Test #15: hydrazine_TestThread ...........................   Passed    0.02 sec
      Start 16: hydrazine_TestTimer
16/18 Test #16: hydrazine_TestTimer ............................   Passed    0.00 sec
      Start 17: hydrazine_TestXmlArgumentParser
17/18 Test #17: hydrazine_TestXmlArgumentParser ................   Passed    0.01 sec
      Start 18: hydrazine_TestXmlParser
18/18 Test #18: hydrazine_TestXmlParser ........................   Passed    0.01 sec

100% tests passed, 0 tests failed out of 18
```

Thanks

**Comment by dmikushin** [comment]
Great, README is corrected in 05a89a93e0831db302780b3e49d97839cb7a64b3, thank you!
